virtual-desktop/src/app/context-menu/context-menu.component.ts

```diff
@@ -20,18 +20,21 @@ import {
   ViewChild,
   ViewChildren,
   AfterViewInit,
-  QueryList
+  QueryList,
+  OnDestroy,
+  OnInit
 } from '@angular/core';
 
 import { ContextMenuItem } from 'pluginlib/inject-resources';
 import { DomSanitizer } from '@angular/platform-browser';
+import { ContextMenuService } from './contextmenu.service';
```

Comment:
I don't think this is a good use of the service. Can you not just add a "showContextMenu" boolean, or something, to the ContextMenuComponent and then use that as a conditional for whether certain keyboard bindings are on or off. The Keybinding service shouldn't be "alerted" when it can use a key again, it should try to use that key and if a case doesn't match, nothing happens for that key press. Using a service is slow here (you can test this by pressing "Right" to open the menu, pressing "Escape" quickly to close, and trying to press "Right" again it won't work) And I understand that this was intentional due to your 1000ms timeout for your subscription, but that's why it's not a good idea. A boolean will also work here because you're never going to have multiple context menu's (or you shouldn't) so keeping track of multiple components isn't a problem here. Because the ContextMenu component won't exist when you don't have it spawned, your initial check might want to be like 
`if (contextMenu && contextMenu.showContextMenu)`

Seeing as right now, it seems contextMenu is only available when it's visible, you can probably get away with just doing the 1st check, but adding that boolean might be a cleaner, easier to see way of doing it

Reply:
Sure, if you think it works i can try this and let you know if it does the job or not, ofcourse simpler the better

Reply:
Sure, that sounds good. Lmk if you need help

Reply:
If it becomes too difficult to import the component, maybe due to module inconsistencies or something, it will probably be fine to use the service, just do it in a way without the 1000ms by doing isShown checks, instead of emitter events

Reply:
removed service altogether and removed 1000ms delay as well, found one way